packages/trpc/server/routers/viewer/organizations/create.handler.ts

```diff
@@ -38,6 +47,93 @@ const getIPAddress = async (url: string): Promise<string> => {
   });
 };
 
+const checkUserIsAdmin = (
+  userRole: LoggedInUserType["role"],
+  loggedInUserEmail: string,
+  orgOwnerEmail: string
+) => {
+  const IS_USER_ADMIN = userRole === UserPermissionRole.ADMIN;
+  if (!ORG_SELF_SERVE_ENABLED && !IS_USER_ADMIN) {
+    throw new TRPCError({ code: "FORBIDDEN", message: "Only admins can create organizations" });
+  }
+
+  if (!IS_USER_ADMIN && loggedInUserEmail !== orgOwnerEmail) {
+    throw new TRPCError({
+      code: "FORBIDDEN",
+      message: "You can only create organization where you are the owner",
+    });
+  }
+
+  return IS_USER_ADMIN;
+};
+
+const checkLoginStatus = async (userId: number) => {
+  const loggedInUser = await prisma.user.findUnique({
+    where: {
+      id: userId,
+    },
+    select: {
+      id: true,
+      role: true,
+      email: true,
+      teams: {
+        select: {
+          team: {
+            select: {
+              slug: true,
+            },
+          },
+        },
+      },
+    },
+  });
+  if (!loggedInUser) throw new TRPCError({ code: "UNAUTHORIZED", message: "You are not authorized." });
+
+  return loggedInUser;
+};
+
+const checkOrgPublishedTeams = (teams: LoggedInUserType["teams"], isAdmin: boolean) => {
+  const publishedTeams = teams.filter((team) => !!team.team.slug);
+  if (!isAdmin && publishedTeams.length < ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE) {
+    throw new TRPCError({ code: "FORBIDDEN", message: "You need to have atleast two published teams." });
```

Comment:
Lets make this a little easier to figure out whats going on 
```suggestion
   if(isAdmin) return;
   
  const publishedTeams = teams.filter((team) => !!team.team.slug);
  if (publishedTeams.length < ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE) {
    throw new TRPCError({ code: "FORBIDDEN", message: "You need to have atleast two published teams." });
```

Reply:
Agreed, this is actually much much better